pkg/autoscaler/statserver/server_test.go

```diff
@@ -166,6 +174,44 @@ func TestServerDoesNotLeakGoroutines(t *testing.T) {
 	server.Shutdown(time.Second)
 }
 
+func BenchmarkStatServer(b *testing.B) {
+	for encoding, jsonEncoding := range map[string]bool{"json": true, "gob": false} {
+		for _, numMsgs := range []int{1, 5, 100} {
+			statsCh := make(chan metrics.StatMessage, numMsgs)
+			server := newTestServer(statsCh)
+			go server.listenAndServe()
```

Comment:
Do we need to `defer server.Shutdown` before error checking in testing?

Reply:
all the errors are fatal, so if anything fails the whole process is going to exit anyway. This matches the behaviour elsewhere in the file.